charts/spinnaker/templates/configmap/halyard-config.yaml

```
@@ -109,9 +109,24 @@ data:
     {{- end }}
 
     $HAL_COMMAND config provider kubernetes enable
-    {{- range $name, $account := .Values.kubeConfig.accounts }}
 
-    if $HAL_COMMAND config provider kubernetes account get {{ $name }}; then
+    function custom_resource_exists() {
+      local account="${1}"
+      local kubernetes_kind="${2}"
+      local spinnaker_kind="${3}"
+      echo "${account}" | grep -q '"kubernetesKind":"'"${kubernetes_kind}"'","spinnakerKind":"'"${spinnaker_kind}"'"'
+    }
+
+    function kubernetes_kind_is_bound() {
+      local account="${1}"
+      local kubernetes_kind="${2}"
+      echo "${account}" | grep -q '"kubernetesKind":"'"${kubernetes_kind}"'"'
+    }
+
+    {{- range $name, $account := .Values.kubeConfig.accounts }}
+    ACCOUNT="$($HAL_COMMAND config provider kubernetes account get {{ $name }} --no-validate --quiet --output=json || exit 0)"
```

It is sad we don't have `jq` available in the Halyard pods :cry: Otherwise we could've added `| head -1` to get only the JSON part and do cleaner `jq` queries in the functions. This would also protect us against changed output of Halyard in future upgrades (e.g. field sorting or newly added fields in between).

```suggestion
    ACCOUNT="$($HAL_COMMAND config provider kubernetes account get {{ $name }} --no-validate --quiet --output=json | head -1 || exit 0)"
```

---

Indeed. This is poor man's `jq` currently. 😄